src/TextField/TextField.jsx

```diff
@@ -370,6 +391,12 @@ const TextField = React.createClass({
     if (this.props.onKeyDown) this.props.onKeyDown(event);
   },
 
+  _handleInputKeyUp() {
+    this.setState({currentCharCount: this.getValue().length});
+    if (this.props.charCountMax >= this.getValue().length) this.setState({charCountError: false});
```

Comment:
@heetvachhani I think coupling underline and label to TextCharCount isn't idea. Hang tight, I think we're going to have to reconsider some of the `TextField` component's design to implement this properly.